src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/atan.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/subtract.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_atan2(const NodeContext& context) {
+    // atan2(input, other, *) → Tensor
+    num_inputs_check(context, 2, 2);
+    Output<Node> lhs;
+    Output<Node> rhs;
+
+    std::tie(lhs, rhs) = get_inputs_with_promoted_types(context, 0, 1);
+
+    auto div = context.mark_node(std::make_shared<v1::Divide>(lhs, rhs));
+
+    auto atan = context.mark_node(std::make_shared<v0::Atan>(div));
+
+    // create some constants to adjust result according to quadrant.
+    auto zero = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {0}));
+    auto pi = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {M_PI}));
+    auto neg_pi = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {-M_PI}));
```

Comment:
`rhs` or `lhs` type can be `dynamic` in that case creation of constant will fail. Please create constant with `i32` or `f64` type and use `ConvertLike` to make it of right dtype.

Reply:
i have made the change. please take a look when you have a moment :)